src/tests/support.cpp

```diff
@@ -497,6 +497,13 @@ test_value_equal(const char *what, const char *expected_value, const uint8_t v[]
     return 0;
 }
 
+bool
+mpi_empty(const pgp_mpi_t &val)
+{
+    pgp_mpi_t zero{};
+    return (val.len == 0) && !memcmp(val.mpi, zero.mpi, PGP_MPINT_SIZE);
```

Comment:
@antonsviridenko  I added this check for the case if, for instance, we'll clear mpi's length but forget to clean up the mpi contents, which would be a major security issue for the secret key's mpi.

And it will not affect performance at all as it is called from the few tests, and takes magnitudes less cycles then the single key generation operation.